packages/remediations/src/steps/reviewSystems.js

```diff
@@ -31,23 +34,40 @@ const ReviewSystems = ({ issues, systems, registry, ...props }) => {
     const formOptions = useFormApi();
     const inventoryApi = useRef({});
 
+    const formValues = formOptions.getState().values;
+    const playbook = formValues[EXISTING_PLAYBOOK];
+    const error = formOptions.getState().errors?.systems;
+
     const rowsLength = useSelector(({ entities }) => (entities?.rows || []).length);
     const selected = useSelector(({ entities }) => entities?.selected || []);
     const loaded = useSelector(({ entities }) => entities?.loaded);
+    const newSystemsNamed = useSelector(({ hostReducer: { hosts } }) => hosts?.map(host => (
+        { id: host.id, display_name: host.display_name })) || []
+    );
 
-    const formValues = formOptions.getState().values;
-    const error = formOptions.getState().errors?.systems;
-    const playbook = formValues && formValues[EXISTING_PLAYBOOK];
+    const allSystems = useRef([]);
+    const [ allSystemsNamed, setAllSystemsNamed ] = useState([]);
 
-    const allNewSystems = dedupeArray(issues.reduce((acc, curr) => [
-        ...acc,
-        ...(curr.systems || [])
-    ], [ ...systems ]));
+    useEffect(() => {
+        const playbookSystems = formValues && formValues[EXISTING_PLAYBOOK_SELECTED]
+            ? getPlaybookSystems(playbook)
+            : [];
+        setAllSystemsNamed(unionWith(playbookSystems, newSystemsNamed, isEqual));
+        const allNewSystems = dedupeArray(issues.reduce((acc, curr) => [
+            ...acc,
+            ...(curr.systems || [])
+        ], [ ...systems ]));
+        allSystems.current = dedupeArray([
+            ...allNewSystems,
+            ...(playbookSystems.map(system => system.id))
+        ]);
```

Comment:
You can theoretically merge these two together, but you are sacrificing readability at that point. So your call @fhlavac 
```JS
        allSystems.current = dedupeArray([
            ...dedupeArray(issues.reduce((acc, curr) => [
              ...acc,
              ...(curr.systems || [])
            ], [ ...systems ])),
            ...(playbookSystems.map(system => system.id))
        ]);
```

Reply:
@karelhala I think we can change it, the names of the variables make it still understandable, thank you